tfhe/src/shortint/server_key/mod.rs

```diff
@@ -951,12 +991,18 @@ impl From<CompressedServerKey> for ServerKey {
             },
         );
 
+        let max_noise_level = MaxNoiseLevel::new(divide_ceil(
+            carry_modulus.0 * message_modulus.0 - 1,
+            message_modulus.0 - 1,
+        ));
```

Comment:
I think we need a formula, the code is repeated 4 or 5 times, it should take a carry modulus and a message modulus

It will allow to add some small tests for e.g. 2_2 like values and check we get a max noise level of 5

Reply:
Put into `MaxNoiseLevel::from_msg_carry_modulus`